File: src/main/java/com/kishlaly/ta/analyze/tasks/blocks/two/Long_ScreenTwo_EMA_LastBarNotAbove.java
package com.kishlaly.ta.analyze.tasks.blocks.two;

import com.kishlaly.ta.model.BlockResult;
import com.kishlaly.ta.model.SymbolData;
import com.kishlaly.ta.model.indicators.EMA;
import com.kishlaly.ta.model.indicators.Indicator;
import com.kishlaly.ta.utils.Log;

import java.util.List;

import static com.kishlaly.ta.analyze.BlockResultCode.LAST_QUOTE_ABOVE_EMA_SCREEN_2;
import static com.kishlaly.ta.analyze.BlockResultCode.OK;
import static com.kishlaly.ta.utils.Quotes.isQuoteAboveEMA;

/**
 * последний столбик не выше ЕМА13
 */
public class Long_ScreenTwo_EMA_LastBarNotAbove implements ScreenTwoBlock {
    @Override
    public BlockResult check(SymbolData screen) {
        List<EMA> screen_2_EMA13 = (List<EMA>) screen.indicators.get(Indicator.EMA13);
        if (isQuoteAboveEMA(screen.getLastQuote(), screen_2_EMA13.get(screen_2_EMA13.size() - 1).getValue())) {
            Log.recordCode(LAST_QUOTE_ABOVE_EMA_SCREEN_2, screen);
            Log.addDebugLine("Последний столбик выше EMA на втором экране");
            return new BlockResult(screen.getLastQuote(), LAST_QUOTE_ABOVE_EMA_SCREEN_2);
        }
        return new BlockResult(screen.getLastQuote(), OK);
    }
}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/blocks/two/Long_ScreenTwo_EMA_LastBarNotAbove.kt
package com.kishlaly.ta.analyze.tasks.blocks.two

import com.kishlaly.ta.model.BlockResult
import com.kishlaly.ta.model.BlockResultCode
import com.kishlaly.ta.model.SymbolData
import com.kishlaly.ta.model.above
import com.kishlaly.ta.model.indicators.EMA
import com.kishlaly.ta.model.indicators.Indicator
import com.kishlaly.ta.utils.Log

/**
 * last bar no higher than EMA13
 */
class Long_ScreenTwo_EMA_LastBarNotAbove : ScreenTwoBlock {
    override fun check(screen: SymbolData): BlockResult {
        val screen_2_EMA13 = screen.indicator(Indicator.EMA13) as List<EMA>
        if (screen.lastQuote above screen_2_EMA13.last()) {
            Log.recordCode(BlockResultCode.LAST_QUOTE_ABOVE_EMA_SCREEN_2, screen)
            Log.addDebugLine("Last bar above the EMA on the second screen")
            return BlockResult(screen.lastQuote, BlockResultCode.LAST_QUOTE_ABOVE_EMA_SCREEN_2)
        }
        return BlockResult(screen.lastQuote, BlockResultCode.OK)
    }
}